AutoBazelRepository in empty package fails

https://github.com/bazelbuild/bazel/blob/652f83e5f0bdca43631cfec8ac5686954fc427ee/tools/java/runfiles/AutoBazelRepositoryProcessor.java#L99

Here is a sample:
```
import com.google.devtools.build.runfiles.AutoBazelRepository;
import com.google.devtools.build.runfiles.Runfiles;
import org.junit.Test;

import java.nio.file.Path;
import java.nio.file.Paths;

@AutoBazelRepository
public class ExampleTest {

    @Test
    public void doNothingTest() throws Exception {
        Runfiles.Preloaded runfiles = Runfiles.preload();
        String path = runfiles.withSourceRepository(AutoBazelRepository_ExampleTest.NAME)
                .rlocation("my_workspace/path/to/my/data.txt");
    }
}
```

This produces a generated class for the following:

```java
package ;

class AutoBazelRepository_ExampleTest {
  /**
   * The canonical name of the repository containing the Bazel target that
   * compiled {@link TestContainerExampleTest}.
   */
  static final String NAME = "";

  private AutoBazelRepository_ExampleTest () {}
}
```

Please notice the `package ;` this fails to compile

```
bazel-out/darwin_arm64-fastbuild/bin/src/_javac/ExampleTest/TExampleTest_sources/AutoBazelRepository_ExampleTest.java:1: error: <identifier> expected
package ;
```

--- a/tools/java/runfiles/AutoBazelRepositoryProcessor.java
+++ b/tools/java/runfiles/AutoBazelRepositoryProcessor.java
@@ -96,7 +96,11 @@ public final class AutoBazelRepositoryProcessor extends AbstractProcessor {
     try (PrintWriter out =
         new PrintWriter(
             processingEnv.getFiler().createSourceFile(generatedClassName).openWriter())) {
-      out.printf("package %s;\n", generatedClassPackage);
+
+      if (!generatedClassPackage.isEmpty()) {
+        // This annotation may exist on a class which is at the root package
+        out.printf("package %s;\n", generatedClassPackage);
+      }
       out.printf("\n");
       out.printf("class %s {\n", generatedClassSimpleName);
       out.printf("  /**\n");

